spring-integration-file/src/main/java/org/springframework/integration/file/filters/LastModifiedFileListFilter.java

```diff
@@ -65,13 +72,22 @@ public LastModifiedFileListFilter(long age) {
 	 * is filtered. The resolution is seconds.
 	 * Defaults to 60 seconds.
 	 * @param age the age
+	 * @param unit the timeUnit.
 	 */
-	public void setAge(long age) {
-		setAge(age, TimeUnit.SECONDS);
+	public void setAge(long age, TimeUnit unit) {
+		this.age = unit.toSeconds(age);
 	}
 
-	public long getAge() {
-		return this.age;
+	/**
+	 * Set the age that files have to be before being passed by this filter.
+	 * If {@link File#lastModified()} plus age is greater than the current time, the file
+	 * is filtered. The resolution is seconds.
+	 * Defaults to 60 seconds.
+	 * @param age the age
+	 * @since 5.1.3
+	 */
+	public void setAge(Duration age) {
+		setAge(age.get(ChronoUnit.SECONDS));
```

Comment:
Why not just `age.getSeconds()` ? Simpler, and functionally the same.

Reply:
Because it is not `public` 😄

Reply:
`toSeconds()` is private, but...

```java
    public long getSeconds() {
        return seconds;
    }
```